plugins/in_splunk/splunk_prot.c

```diff
@@ -0,0 +1,822 @@
+    }
+
+    /* Check if we have a Host header: Hostname ; port */
+    mk_http_point_header(&request->host, &session->parser, MK_HEADER_HOST);
+
+    /* Header: Connection */
+    mk_http_point_header(&request->connection, &session->parser,
+                         MK_HEADER_CONNECTION);
+
+    /* HTTP/1.1 needs Host header */
+    if (!request->host.data && request->protocol == MK_HTTP_PROTOCOL_11) {
+        flb_sds_destroy(tag);
+        mk_mem_free(uri);
+
+        return -1;
+    }
+
+    /* Should we close the session after this request ? */
+    mk_http_keepalive_check(session, request, ctx->server);
+
+    /* Content Length */
+    header = &session->parser.headers[MK_HEADER_CONTENT_LENGTH];
+    if (header->type == MK_HEADER_CONTENT_LENGTH) {
+        request->_content_length.data = header->val.data;
+        request->_content_length.len  = header->val.len;
+    }
+    else {
+        request->_content_length.data = NULL;
+    }
+
+    if (request->method == MK_METHOD_GET) {
+        /* Handle health minotoring of splunk hec endpoint for load balancers */
+        if (strncmp(uri, "/services/collector/health", 26) == 0) {
+            send_json_message_response(conn, 200, "{\"text\":\"Success\",\"code\":200}");
+        }
+        else {
+            send_response(conn, 400, "error: invalid HTTP endpoint\n");
+        }
+
+        flb_sds_destroy(tag);
+        mk_mem_free(uri);
+
+        return 0;
+    }
+
+    /* Under services/collector endpoints are required for
+     * authentication if provided splunk_token */
+    ret = validate_auth_header(ctx, request);
+    if (ret < 0){
+        send_response(conn, 401, "error: unauthroized\n");
+        if (ret == SPLUNK_AUTH_MISSING_CRED) {
+            flb_plg_warn(ctx->ins, "missing credentials in request headers", ret);
+        }
+        else if (ret == SPLUNK_AUTH_UNAUTHORIZED) {
+            flb_plg_warn(ctx->ins, "wrong credentials in request headers", ret);
+        }
+
+        flb_sds_destroy(tag);
+        mk_mem_free(uri);
+
+        return -1;
+    }
+
+    if (request->method == MK_METHOD_POST) {
+        if (strncmp(uri, "/services/collector/event", 25) == 0 ||
+            (strncmp(uri, "/services/collector", 19) == 0 && strlen(uri) == 19)) {
+            ret = process_hec_payload(ctx, conn, tag, session, request);
+
+            send_json_message_response(conn, 200, "{\"text\":\"Success\",\"code\":0}");
+
+            flb_sds_destroy(tag);
+            mk_mem_free(uri);
+        }
+        else if (strncmp(uri, "/services/collector/raw", 23) == 0) {
+            ret = process_hec_raw_payload(ctx, conn, tag, session, request);
+
+            send_json_message_response(conn, 200, "{\"text\":\"Success\",\"code\":0}");
+
+            flb_sds_destroy(tag);
+            mk_mem_free(uri);
+        }
+        else {
+            send_response(conn, 400, "error: invalid HTTP endpoint\n");
+            return -1;
+        }
+    }
+
+    if (request->method != MK_METHOD_GET &&
```

Comment:
This needs to be an `else` to the previous block. We tested for `GET` in line 743.

Reply:
Got it. Your suggestion simplifies the validation of HTTP methods.
Fixed in https://github.com/fluent/fluent-bit/pull/7349/commits/6195e16de5d8ee16b123d2568157e0e8bca1611c.